### PR TITLE
fix: Failed to open X display when restarting container

### PR DESCRIPTION
**Pull request by a14stoner**
fix error:

what(): Failed to open X display

by deleting the .X99-lock file at the start of the container

### CODE REVIEW

**Comment by acalcutt** [comment]
Could you put in an issue with steps to reproduce the issue you are having? I'm not sure about deleting a lockfile as a solution since would since wouldn't xvfb that made it still be running?

**Comment by a14stoner** [comment]
Hey @acalcutt -> #794 created this issue.

**Comment by acalcutt** [comment]
I tested this fix and it does work, but I'm not sure I like the syntax of this file removal. I feel like it should check if the display exists before trying to delete the file, similar to https://github.com/maptiler/tileserver-gl/pull/802

I tested this like this
```
root@mapgen2:/opt/data_test# docker build -t 105 -f Dockerfile .
root@mapgen2:/opt/data_test# docker run -dt --name 105-2 -h tileserver-gt -v $(pwd):/data -p 9999:8080 105:latest
d2e9f7e488d5a927bd911b5250dd205535a9571c622cc744af45b9b658651dd9
root@mapgen2:/opt/data_test# docker restart 105-2
105-2
root@mapgen2:/opt/data_test# docker container logs 105-2
Starting tileserver-gl v4.4.9
No MBTiles specified, using output.mbtiles
[INFO] Automatically creating config file for output.mbtiles
[INFO] Only a basic preview style will be used.
[INFO] See documentation to learn how to create config.json file.
Run with --verbose to see the config file here.
Starting server
Listening at http://[::]:8080/
Startup complete
Caught signal SIGTERM, stopping gracefully
Starting tileserver-gl v4.4.9
No MBTiles specified, using output.mbtiles
[INFO] Automatically creating config file for output.mbtiles
[INFO] Only a basic preview style will be used.
[INFO] See documentation to learn how to create config.json file.
Run with --verbose to see the config file here.
Starting server
Listening at http://[::]:8080/
Startup complete
GET /styles/basic-preview/2/2/3.png 200 3804 - 302.011 ms
GET /styles/basic-preview/2/0/3.png 200 6029 - 299.944 ms
GET /styles/basic-preview/2/0/0.png 200 7931 - 303.400 ms
GET /styles/basic-preview/2/3/0.png 200 5209 - 19.968 ms
GET /styles/basic-preview/2/3/1.png 200 28328 - 23.855 ms
GET /health 200 2 - 0.541 ms

```

File: docker-entrypoint.sh [hunk 1]
```diff
@@ -1,6 +1,7 @@
 #!/bin/sh
 if ! which -- "${1}"; then
   # first arg is not an executable
+  rm /tmp/.X99-lock -f > /dev/null || ``
```
Comment (acalcutt) [inline docker-entrypoint.sh:4]: why does this need the `|| ``` part. seems like whether this file exits or not it will be deleted with no messaage.

Comment (a14stoner) [reply docker-entrypoint.sh:4]: Thats like a Try-Catch :D When the file doesnt exist and i dont have the 2 pipes in it the container wont start.

Comment (a14stoner) [reply docker-entrypoint.sh]: i updated the docker-entrypoint.sh

now the deletion is only executed when the file exists.

Comment (acalcutt) [reply docker-entrypoint.sh]: I like this better, could we make it a single line, like
`if [ -e /tmp/.X99-lock ]; then rm /tmp/.X99-lock -f; fi`

**Comment by acalcutt** [comment]
If you make this one change I will merge this
https://github.com/maptiler/tileserver-gl/pull/790#discussion_r1141094590

**Comment by a14stoner** [comment]
Done 🙋🏼‍♂️

**Review by acalcutt — APPROVED** [review]

**Pull request merged**